### PR TITLE
Update unix gpu toolchain

### PR DESCRIPTION
**Pull request by ChaiBapchya**
## Description ##
Currently, Unix GPU & Centos GPU tests use P3 & G3 AWS EC2 instances.
In an effort to improve the cost & efficiency, switch to G4 EC2 instances has been proposed.

This switch involves upgrading the GPU toolchain broadly
| Host Machine | Old | New |
|-------------------|---------|-----------|
| Ubuntu LTS | 16.04.3 | 18.04.3 |
| Tesla Driver |   M60 | T4 |
| EC2 Instance Type | G3 | G4 |
| Docker | 18.09 | 19.03 |
| NVidia Driver | 418.56 | 440.33.01 |
| Cuda Driver | 10.1 | 10.2 |

### Code Changes ###
1. Latest Docker [19.03] has built-in cuda support [hence replace nvidia-docker with docker --gpus all]
2. Given that the host machine has updated drivers, TVM Op shouldn't need cuda compat [`/usr/local/cuda/compat`]
3. replacing `ubuntu_gpu_cu101` with `ubuntu_build_cuda`
Docker compose follows multi-stage build [https://docs.docker.com/develop/develop-images/multistage-build/] and defines multiple targets
`ubuntu_build_cuda` target is `gpuwithcudaruntimelibs`
`ubuntu_gpu_cu101`  target is : gpuwithcompatenv [which has been commented out now]

4. ~~After testing this on CI Dev account : http://jenkins.mxnet-ci-dev.amazon-ml.com/blue/organizations/jenkins/mxnet-validation-bapac%2Funix-gpu/detail/update_gpu_toolchain/8/pipeline~~
~~The TVMOpError related to Binary Ops was encountered : https://github.com/apache/incubator-mxnet/issues/17840~~
~~To unblock the migration from G3 to G4, these flaky tests have been skipped.~~

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- [x] Code is well-documented: 
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

## Comments ##
Thanks to @ptrendx for the help identifying libcuda compat as the rootcause for 
```
CUDA: Check failed: e == cudaSuccess (803 vs. 0) : system has unsupported display driver / cuda driver combination
```
Helped me close : https://github.com/NVIDIA/nvidia-docker/issues/1256

Thanks to @leezu and @josephevans throughout this migration effort and @sandeep-krishnamurthy @szha for the guidance.

### CODE REVIEW

File: ci/jenkins/Jenkins_steps.groovy [hunk 1]
```diff
@@ -155,7 +155,7 @@ def compile_unix_int64_gpu() {
         ws('workspace/build-gpu-int64') {
           timeout(time: max_time, unit: 'MINUTES') {
             utils.init_git()
-            utils.docker_run('ubuntu_gpu_cu101', 'build_ubuntu_gpu_large_tensor', false)
+            utils.docker_run('ubuntu_build_cuda', 'build_ubuntu_gpu_large_tensor', false)
```
Comment (leezu) [inline ci/jenkins/Jenkins_steps.groovy:158]: Why this change? Note that `ubunt_build_cuda` is an image that is only provided to workaround some broken design problems

https://github.com/apache/incubator-mxnet/blob/76fa58373636c57fee1e4e6cd7960723b39f455f/ci/docker/Dockerfile.build.ubuntu#L153-L162

We shouldn't migrate more tests to use this image, but rather get rid of this image

Comment (ChaiBapchya) [reply ci/jenkins/Jenkins_steps.groovy:158]: So the aim is to get rid of both ubuntu_build_cuda and ubuntu_cu101 and instead use `ubuntu_cpu` [base image]
Correct? @leezu

Comment (leezu) [reply ci/jenkins/Jenkins_steps.groovy:158]: Only `ubuntu_build_cuda`, as we still need

https://github.com/apache/incubator-mxnet/blob/76fa58373636c57fee1e4e6cd7960723b39f455f/ci/docker/Dockerfile.build.ubuntu#L143-L150

Comment (leezu) [reply ci/jenkins/Jenkins_steps.groovy:158]: ubuntu_cu101, ubuntu_cu102, etc all make use of that `gpu` target, but have a different base image

**Comment by ChaiBapchya** [comment]
Rebased to fix windows-gpu issue : fixed in https://github.com/apache/incubator-mxnet/pull/18177

File: ci/jenkins/Jenkins_steps.groovy (hunk 1 above)
Comment (ChaiBapchya) [reply ci/jenkins/Jenkins_steps.groovy:158]: Wait why not get rid of ubuntu_cu101 as well?

Comment (ChaiBapchya) [reply ci/jenkins/Jenkins_steps.groovy:158]: We don't want to use libcuda compat right? whereas target of ubuntu_cu101 is gpuwithcompatenv

Comment (leezu) [reply ci/jenkins/Jenkins_steps.groovy:158]: We should get rid of the `gpuwithcompatenv` target and point `ubuntu_cu101` to the `gpu` target.

Comment (leezu) [reply ci/jenkins/Jenkins_steps.groovy:158]: You still need some gpu image. We may later find that (once cuda 11 is released), the gpu target may not be neede anymore and we can point `ubuntu_gpu_cu101` to the `base` target.

`ubuntu_gpu_cu101 would still have a different base image compared to `ubuntu_cpu` `(`nvidia/cuda:10.1-cudnn7-devel-ubuntu18.04` vs `ubuntu:18.04`).

Comment (ChaiBapchya) [reply ci/jenkins/Jenkins_steps.groovy:158]: Added as recommended
https://github.com/apache/incubator-mxnet/pull/18186/commits/022e135585cd830d2095e6611e13afe6955d21ce
Take a look @leezu

Comment (ChaiBapchya) [reply ci/jenkins/Jenkins_steps.groovy]: Oops didn't do it cleanly. Fixing it now. https://github.com/apache/incubator-mxnet/pull/18186/commits/ec5330d7c86e770687053480be585a75712ce2a4

File: tests/python/unittest/test_numpy_op.py [hunk 2]
```diff
@@ -2220,6 +2222,7 @@ def hybrid_forward(self, F, x):
                 assert same(ret_mx.asnumpy(), ret_np)
 
 
+@unittest.skip("Flaky test https://github.com/apache/incubator-mxnet/issues/17840")
```
Comment (leezu) [inline tests/python/unittest/test_numpy_op.py:2225]: TVMOP is always turned off for GPU builds on CI now. You can remove these

Comment (ChaiBapchya) [reply tests/python/unittest/test_numpy_op.py:2225]: Alright, removing all the flaky test skips then

**Comment by ChaiBapchya** [comment]
@mxnet-bot run ci [windows-gpu] 

assertion failed for test_np_mixed_precision_binary_funcs : Likely flaky

**Comment by ChaiBapchya** [comment]
Infra related changes : https://github.com/apache/incubator-mxnet-ci/pull/20
Specifically

- Updated auto-scaling lambda with additional instance node label : mxnetlinux-gpu-g4
- Update env variables corresponding to the instances with G4 node label specific information
```
    LAUNCH_TEMPLATES : "mxnetlinux-gpu-g4":{"id":"lt-0ebf575cc5a56ebf4","version":"1"}
    EXECUTORS_PER_LABEL : "mxnetlinux-gpu-g4":1
    WARM_POOL_SIZE : mxnetlinux-gpu-g4":0
    MINIMUM_QUEUE_TIMES_SEC : mxnetlinux-gpu-g4":30
    CCACHE_EFS_DNS : "mxnetlinux-gpu-g4":"NONE"
    MAXIMUM_STARTUP_TIME_SEC : "mxnetlinux-gpu-g4":300
    MANAGED_JENKINS_NODE_LABELS : "mxnetlinux-gpu-g4"
```
Specific commit : https://github.com/apache/incubator-mxnet-ci/pull/20/commits/1a537af85a1de9135c330d18cb0dcfc9fb6de802

Manually, created a launch template for G4 node pointing to the AMI [created in dev accessible to prod account] [followed steps mentioned here : https://cwiki.apache.org/confluence/display/MXNET/Setup#Setup-Slave]

**Comment by ChaiBapchya** [comment]
`test_np_mixed_precision_binary_funcs`
which previously failed on windows-gpu `Python 3: GPU Win` on multiple occasions
http://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/mxnet-validation%2Fwindows-gpu/detail/PR-18186/8/pipeline
http://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/mxnet-validation%2Fwindows-gpu/detail/PR-18186/7/pipeline

Now fails on unix-gpu :  `Python3: MKLDNN-GPU-NOCUDNN`
http://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/mxnet-validation%2Funix-gpu/detail/PR-18186/8/pipeline/387

**Pull request merged**